clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/security/DcosAccountCredentials.groovy

```diff
@@ -0,0 +1,157 @@
+
+  final String name
+  final String account
+  final String environment
+  final String accountType
+  final List<LinkedDockerRegistryConfiguration> dockerRegistries
+  final List<String> requiredGroupMembership
+  final List<DcosRegion> regions
+  // Not really a fan of creating this just for use within deck, but it works for now
+  final List<DcosClusterInfo> dcosClusters
+  final DcosCredentialMap dcosClusterCredentials
+
+  DcosAccountCredentials(String account,
+                         String environment,
+                         String accountType,
+                         List<LinkedDockerRegistryConfiguration> dockerRegistries,
+                         List<String> requiredGroupMembership,
+                         List<DcosClusterCredentials> clusters) {
+    this.name = account
+    this.account = account
+    this.environment = environment
+    this.accountType = accountType
+    this.dockerRegistries = dockerRegistries != null ? dockerRegistries : new ArrayList<>()
+    this.requiredGroupMembership = requiredGroupMembership
+    this.dcosClusterCredentials = new DcosCredentialMap(clusters)
+    this.dcosClusters = clusters.collect({ new DcosClusterInfo(it.name, it.dockerRegistries)})
+    this.regions = clusters.collect({new DcosRegion(it.name)})
+  }
+
+  static Builder builder() {
+    return new Builder()
+  }
+
+  @JsonIgnore
+  @Override
+  DcosCredentialMap getCredentials() {
+    dcosClusterCredentials
+  }
+
+  @JsonIgnore
+  DcosClusterCredentials getCredentialsByCluster(String cluster) {
+    dcosClusterCredentials.getCredentialsByCluster(cluster)
+  }
+
+  @Override
+  String getCloudProvider() {
+    CLOUD_PROVIDER
+  }
+
+  static class Builder {
+    private String account
+    private String environment
+    private String accountType
+    private List<LinkedDockerRegistryConfiguration> dockerRegistries
+    private List<String> requiredGroupMembership
+    private List<DcosClusterCredentials> clusters
+
+    Builder account(String account) {
+      this.account = account
+      return this
+    }
+
+    Builder environment(String environment) {
+      this.environment = environment
+      return this
+    }
+
+    Builder accountType(String accountType) {
+      this.accountType = accountType
+      return this
+    }
+
+    Builder dockerRegistries(List<LinkedDockerRegistryConfiguration> dockerRegistries) {
+      this.dockerRegistries = dockerRegistries
+      return this
+    }
+
+    Builder requiredGroupMembership(List<String> requiredGroupMembership) {
+      this.requiredGroupMembership = requiredGroupMembership
+      return this
+    }
+
+    Builder clusters(List<DcosClusterCredentials> clusters) {
+      this.clusters = clusters
+      return this
+    }
+
+    DcosAccountCredentials build() {
+      if (!account) {
+        throw new IllegalArgumentException("Account name for DC/OS provider is missing.")
+      }
+
+      if (!MarathonPathId.isPartValid(account)) {
```

Comment:
Since we're including the cluster name as part of the path, we should validate that it's a valid marathon part. All the more reason we should try to get cluster out of the path I guess!